### ISSUE TITLE
Browser 

### ISSUE THREAD

**Issue by audiomuze** [pawelsalawa/sqlitestudio#3561]
### Details
Sometimes it's easier to do a manual cleanup/ edit of table data in the sqlitestudio browser than running sqlite update queries.  sqlitestudio already generally does a great job at this, however, the ability to simultaneously add/edit data to a single or multiple columns across rows a user has selected would provide a huge productivity gain.  An illustration:

![image](https://user-images.githubusercontent.com/6852511/63838467-4d7e7300-c975-11e9-8fbf-44696cacec8d.png)

In the above column there are inconsistencies in the data.  At very quick way to be able to achieve consistency (without running a query or using copy/paste) would be to enable the following behaviour:

- User selects records with good data and bad data (see above pic)

-  a <ctrl-double click> to enter edit mode on whichever one is highlighted by the mouse cursor and allowing the user to edit the selected field (whilst keeping the rest selected [this functionality/ behaviour already exists at present])

- then when the user presses <Enter> to accept their edit, automatically cascade that edit/change to all selected rows/columns.

**Comment by pawelsalawa** [comment]
You can edit one cell, quit edition mode. Select that cell, copy its value, then select all other cells you want to have the same value and paste the value to all of them at once.

Wouldn't that solve your problem?

**Comment by audiomuze** [comment]
Yes, I do that already, but it's a lot more steps than just highlighting all you want affected, making the change and pressing enter.

Install puddletag and give it a try and you'll see why it's so useful for bulk editing.  Say you have 50 rows so with good data some with bad data and the good and the bad is dispersed throughout the list - it's a lot easier to select the range, edit one good one and hit enter to have it cascade.  It's a real time and trouble saver if implemented.

**Comment by pawelsalawa** [comment]
I'm not sure if it's easier than just ctrl+c on correct value, then select all cells and hit ctrl+v. That sounds much simpler to me. Isn't it to you?

**Comment by audiomuze** [comment]
> I'm not sure if it's easier than just ctrl+c on correct value, then select all cells and hit ctrl+v. That sounds much simpler to me. Isn't it to you?

Intuitively no, but let me play with that and give you honest feedback.  I think the thing that will remain missing is the ability to select a range of "cells" and just start typing or hitting ctrl-v to paste clipboard content into the cell being edited and then having that cascade across all selection when hitting enter.

**Comment by pawelsalawa** [comment]
Well, that could turn out a very surprising to some people, when upon committing edition, all selected cells change their value, since they edited only one. This would have to be more distinguishable. For example a right-click menu entry to enter "multi-cell edition" instead of single cell edition, which is by default (and the only available mode at the moment).

But yes, let's keep this post for tracking, as I see no problem with adding such option in context menu and also to have configuration option to make it default (to make it work you described, but only when enabled in configuration, to avoid surprise and unintended editions by current users).

**Comment by audiomuze** [comment]
Sorry, I don't think I was clear enough.  This is how the behaviour works in puddletag:

If you've selected a series of "cells" (you're in effect doing spreadsheet type editing)
- if you start typing all selections remain selected whilst you edit the last cell selected, overwriting its contents with what you've started typing. If you press F2 instead of just start typing then you're put into edit mode on the last cell from the selection, whilst all selections remain selected.  If you press Esc the edit is abandoned.  If you press Enter it is committed to that cell and all other cells you selected.
- if after you've selected the cells you then Ctrl-Double-Click in any one of the cells in the selection you then enter Edit mode for that cell whilst all selections remain selected.  You can then simply hit Enter to accept that cell's value and cascade it across all the cells you've selected - so a very quick way to change all values in a selection to another cell's pre-existing value.

I think the above behaviours (whilst adding efficiency gains for quick editing) are intuitive - why else would you pre-select multiple cells before editing a cell's contents?  The rollback cells feature would in any event still be available in the event of user error, so no real risk.

Finally, unless I'm overlooking something there's no clash at all with existing editing methods - so no need to have the behavours hidden in a sub-menu (it'll kill the intuitiveness and speed).  Users will only become aware of the capability if they read the release notes and once they know about it I'm pretty sure they'll use it extensively if doing manual updates based on browsing query results

**Comment by pawelsalawa** [comment]
You conveinced me :) Indeed it should not be harmful for existing users and I can see how sometime it can be more handy than copy pasting the way I described it earlier.

I added it for 3.3.0 (the next release with new features, as 3.2.2 is to be bugfix release).

**Comment by audiomuze** [comment]
Great to hear, thank you.
